mpolyakov/homework_4/olx_scraping/olx_scraping/spiders/product_spider.py

```diff
@@ -0,0 +1,68 @@
+import scrapy
+from scrapy.exceptions import CloseSpider
+
+from olx_scraping.items import Product
+
+
+class ProductSpider(scrapy.Spider):
+    name = 'products'
+
+    custom_settings = {
+        'FEED_URI': 'output.json',
+        'FEED_EXPORT_ENCODING': 'utf-8',
+        'CLOSESPIDER_ITEMCOUNT': 10,
+    }
+
+    allowed_domains = ['olx.ua']
+    start_urls = ['https://www.olx.ua/elektronika/']
+
+    MAX_COUNT = 10  # defines maximum items to store
+    count = 0
+
+    def parse(self, response):
+        """
+        Find links to product page.
+        """
+        product_page_links = response.css('.detailsLink')
+        yield from response.follow_all(product_page_links, self.parse_product)
+
+        pagination_links = response.css('span.fleft a')
+        yield from response.follow_all(pagination_links, self.parse)
+
+    def parse_product(self, response):
+        """
+        Extract data from product page
+        """
+        def extract_with_css(query):
+            return response.css(query).get(default='').strip()
+
+        if self.count < self.MAX_COUNT:
+            product = Product()
+            product['name'] = extract_with_css('h1::text')
+            product['url'] = response.xpath('/html/head/link[1]/@href').get()
```

Comment:
I'm sure that it is possible to extract URL from response object without using css or xpath